daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/DaffodilInputContentHandler.scala

```diff
@@ -0,0 +1,250 @@
+ * sent to the inputter, it signals the end of events coming from the contentHandler. This
+ * ends the unparseProcess and returns the event with the unparseResult and/or any error
+ * information
+ *
+ * @param dp dataprocessor object that will be used to call the parse
+ * @param output outputChannel of choice where the unparsed data is stored
+ */
+class DaffodilInputContentHandler(
+  dp: DFDL.DataProcessor,
+  output: DFDL.Output)
+  extends DFDL.DaffodilInputContentHandler {
+  private val inputter = new SAXInfosetInputter(this, dp, output)
+  private var unparseResult: DFDL.UnparseResult = _
+  private val infosetEvent: DFDL.SaxInfosetEvent = new DFDL.SaxInfosetEvent
+  private val characterData = new StringBuilder()
+  private var prefixMapping: NamespaceBinding = _
+
+  def getUnparseResult: DFDL.UnparseResult = unparseResult
+
+  def enableInputterUriAbsolutization(): Unit = inputter.enableUriAbsolutization = true
+
+  override def setDocumentLocator(locator: Locator): Unit = {
+    // do nothing
+  }
+
+  override def startDocument(): Unit = {
+    infosetEvent.eventType = One(StartDocument)
+    sendToInputter()
+  }
+
+  override def endDocument(): Unit = {
+    infosetEvent.eventType = One(EndDocument)
+    sendToInputter()
+  }
+
+  override def startPrefixMapping(prefix: String, uri: String): Unit = {
+    val pre = if (prefix == "") null else prefix
+    prefixMapping = NamespaceBinding(pre, uri, prefixMapping)
+  }
+
+  override def endPrefixMapping(prefix: String): Unit = {
+    // do nothing
+  }
+
+  override def startElement(uri: String, localName: String, qName: String, atts: Attributes): Unit = {
+    // we need to check if the characters data is all whitespace, if it is we drop the whitespace
+    // data, if it is not, it is an error as starting a new element with actual characterData means
+    // we haven't hit an endElement yet, which means we're in a complexElement and a complexElement
+    // cannot have character content
+    if (characterData.nonEmpty && characterData.toString().trim.nonEmpty) {
+      throw new IllegalContentWhereEventExpected("Non-whitespace characters in complex " +
+        "Element: " + characterData.toString()
+      )
+    } else {
+      // reset since it was whitespace only
+      characterData.setLength(0)
+    }
+
+    if (!infosetEvent.isEmpty && infosetEvent.localName.isDefined) {
+      // we started another element while we were in the process of building a startElement
+      // this means the first element was complex and we are ready for the inputter queue
+      sendToInputter()
+    }
+    // use Attributes to determine xsi:nil value
+    val nilIn = atts.getIndex(XMLConstants.W3C_XML_SCHEMA_INSTANCE_NS_URI, "nil")
+    infosetEvent.nilValue = if (nilIn >= 0) {
+      val nilValue = atts.getValue(nilIn)
+      One(nilValue)
+    } else {
+      Nope
+    }
+    // set localName and namespaceURI
+    val qNameArr = qName.split(":")
```

Comment:
I'm wondering for performance reasons we could avoid this split since it isn't always necessary? I belive split is implemtend via regex, so it's going to be relatively slow. If I'm reading the SAX API, we should either get uri + llocalName or qName. So if we get uri + localName, we can just ignore qName and never have to do a split. Or perhaps just make the qNameArr a lazy val so we only evaluate it if it's needed.